### PR TITLE
[back-port][v0.8] fix: return error when verify empty cert chain

### PR DESCRIPTION
**Pull request by Taowyoo**
back-port https://github.com/fortanix/rust-mbedtls/pull/308 to 0.8.X

### CODE REVIEW

**Comment by Taowyoo** [comment]
bors try

File: mbedtls/src/x509/certificate.rs [hunk 1]
```diff
@@ -1416,4 +1471,5 @@ cYp0bH/RcPTC0Z+ZaqSWMtfxRrk63MJQF9EXpDCdvQRcTMD9D85DJrMKn8aumq0M
         assert!(crate::tests::TestTrait::<dyn Sync, MbedtlsBox<Certificate>>::new().impls_trait(), "MbedtlsBox<Certificate> should be Sync");
         assert!(crate::tests::TestTrait::<dyn Sync, MbedtlsList<Certificate>>::new().impls_trait(), "MbedtlsList<Certificate> should be Sync");
     }
+
```
Comment (raoulstrackx) [inline mbedtls/src/x509/certificate.rs:1474]: Spurious newline

**Review by xinyufort — APPROVED** [review]
Approving as per discussion on https://fortanix.slack.com/archives/CK39ZV0KA/p1693421657593469

**Pull request merged**